### ISSUE TITLE
Error - Sequence length 87 exceeds maximum 80

### ISSUE THREAD

**Issue by anshumanpatil** [tekaratzas/RustGPT#8]
When trying to use HF dataset - https://huggingface.co/datasets/its-myrto/fitness-question-answers/tree/main

Getting below error ( Tried adjusting `MAX_SEQ_LEN` with no luck )

`=== MODEL INFORMATION ===
Network architecture: Embeddings, TransformerBlock, TransformerBlock, TransformerBlock, OutputProjection

=== BEFORE TRAINING ===
Input: User: How do mountains form?
Output: broader broader broader lengthening dehydrated lengthening training obese neurons broader broader broader broader avocado unsu
pportive neurons broader broader hip these neurons broader remains mimic hip neurons broader turn neurons broader neurons remains dehy
drated seeing dehydrated broader broader neurons broader broader neurons unstable stretched directly broader broader hollow both neuro
ns broader broader neurons broader broader seeing started remains seeing unsupportive neurons broader broader broader broader seeing b
roader broader broader seeing broader seeing broader broader

=== PRE-TRAINING MODEL ===
Pre-training on 965 examples for 100 epochs with learning rate 0.0005

thread 'main' panicked at src\embeddings.rs:68:13:
Sequence length 87 exceeds maximum 80
note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
error: process didn't exit successfully: target\debug\llm.exe (exit code: 101)`

**Comment by tekaratzas** [comment]
Nice! Adding the HF datatsets!

Yep time to raise the max token length. Just a const to increase.

The one in main.rs (we should prob remove the ones in lib.rs)

const MAX_SEQ_LEN: usize = 80; <-- we should bring this up. Maybe 100? looks like it crashed with a sequence of length 87

My thoughts in general are to move the size of the model up slowly as we add data to really see how the data impacts it.

**Comment by anshumanpatil** [comment]
Thanks!

I made progress with adjusting `MAX_SEQ_LEN` now planning to move all these in separate .env because when I increase MAX_SEQ_LEN it takes little more time for training. as well saving model is also needs to be planned so the model can get evolved with every new dataset/iteration.

**Comment by tekaratzas** [comment]
Are you running with cargo run --release? Should only take max 15 seconds
